rustvmm_gen/rustvmm_gen.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+#
+# Copyright 2025 © Institute of Software, CAS. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+from pathlib import Path
+from lib_kernel_source import (
+    check_kernel_version,
+    create_temp_dir,
+    download_kernel,
+    extract_kernel,
+    install_headers,
+)
+from lib_syscall import (
+    generate_syscall_table,
+    generate_rust_code,
+)
+
+# Map arch used in linux kernel to arch understandable for Rust
+MAP_RUST_ARCH = {"arm64": "aarch64", "x86_64": "x86_64", "riscv": "riscv64"}
+
+
+def prepare_command(args):
+    try:
+        exists, msg = check_kernel_version(args.version)
+        if not exists:
+            raise RuntimeError(f"Version check failed: {msg}")
+
+        # Create `temp_dir` under `/tmp`
+        temp_dir = create_temp_dir(args.version)
+
+        # Download kernel tarball from https://cdn.kernel.org/
+        tarball = download_kernel(args.version, temp_dir)
+
+        # Extract kernel source
+        src_dir = extract_kernel(tarball, temp_dir)
+
+        # Get headers of specific architecture
+        success = install_headers(
+            src_dir=src_dir,
+            arch=args.arch,
+            install_path=args.install_path,
+        )
+
+        if not success:
+            raise RuntimeError("Installation process failed")
+
+        print(f"\nSuccessfully installed kernel headers to {args.install_path}")
+        return src_dir
+
+    except Exception as e:
+        print(f"\nError: {str(e)}")
+        exit(1)
```

Comment:
Not really a need for this try-catch + exit, I think. The call to this function is the final statement in `main()`, so we hit an exception we'd exit with non-zero code anyway (and additionally printing "Error:" doesn't really add much imo), so can just drop this.